lib/boom_notifier/error_info.ex

```diff
@@ -2,56 +2,55 @@ defmodule ErrorInfo do
   @moduledoc false
 
   # The ErrorInfo struct holds all the information about the exception.
-  # It includes the error message, the stacktrace and context information
+  # It includes the error message, the stacktrace, context information
   # (information about the request, the current controller and action,
-  # among other things).
+  # among other things) and custom data depending on the configuration.
 
   @enforce_keys [:reason, :stack, :timestamp]
-  defstruct [:name, :reason, :stack, :controller, :action, :request, :timestamp]
+  defstruct [:name, :reason, :stack, :controller, :action, :request, :timestamp, :metadata]
+
+  @type option ::
+          :logger
+          | [logger: [fields: list(atom())]]
+          | :assigns
+          | [assigns: [fields: list(atom())]]
+  @type custom_data_strategy_type :: :nothing | option | [option]
```

Comment:
This could also be renamed to `metadata_strategy_type`

Reply:
I like the way it is because `custom_data` is more descriptive to manage internally, then we have "the interface" that is `metadata`. Anyway, I'm ok with changing it as well so let's do it